pkg/controllers/selection/suite_test.go

```diff
@@ -76,6 +76,99 @@ var _ = AfterEach(func() {
 	ExpectProvisioningCleanedUp(ctx, env.Client, provisioners)
 })
 
+var _ = Describe("Namespace Selector", func() {
+	It("should schedule if there is no namespace selector", func() {
+		provisioner.Spec.NamespaceSelector = nil
+		ExpectProvisioned(ctx, env.Client, selectionController, provisioners, provisioner)
+		pod := ExpectProvisioned(ctx, env.Client, selectionController, provisioners, provisioner,
+			test.UnschedulablePod(),
+		)[0]
+		ExpectScheduled(ctx, env.Client, pod)
+	})
+	It("should schedule if there is an empty namespace selector", func() {
+		provisioner.Spec.NamespaceSelector = &metav1.LabelSelector{
+			MatchLabels:      map[string]string{},
+			MatchExpressions: []metav1.LabelSelectorRequirement{},
+		}
+
+		ExpectProvisioned(ctx, env.Client, selectionController, provisioners, provisioner)
+		pod := ExpectProvisioned(ctx, env.Client, selectionController, provisioners, provisioner,
+			test.UnschedulablePod(),
+		)[0]
+		ExpectScheduled(ctx, env.Client, pod)
+	})
+	It("should not schedule if the pod isn't in a matching namespace, MatchLabels", func() {
+		provisioner.Spec.NamespaceSelector = &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"foo": "bar",
+			},
+		}
+		ExpectProvisioned(ctx, env.Client, selectionController, provisioners, provisioner)
+		pod := ExpectProvisioned(ctx, env.Client, selectionController, provisioners, provisioner,
+			test.UnschedulablePod(),
+		)[0]
+		ExpectNotScheduled(ctx, env.Client, pod)
+	})
+	It("should not schedule if the pod isn't in a matching namespace, MatchExpressions", func() {
+		provisioner.Spec.NamespaceSelector = &metav1.LabelSelector{
+			MatchExpressions: []metav1.LabelSelectorRequirement{
+				{
+					Key:      "foo",
+					Operator: metav1.LabelSelectorOpIn,
+					Values:   []string{"bar"},
+				},
+			},
+		}
+		ExpectProvisioned(ctx, env.Client, selectionController, provisioners, provisioner)
+		pod := ExpectProvisioned(ctx, env.Client, selectionController, provisioners, provisioner,
+			test.UnschedulablePod(),
+		)[0]
+		ExpectNotScheduled(ctx, env.Client, pod)
+	})
+	It("should schedule if the pod is in a matching namespace, MatchLabels", func() {
+		ns := randomdata.Noun() + randomdata.Adjective() // need a lowercase name here
+		ExpectCreated(ctx, env.Client, &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   ns,
+				Labels: map[string]string{"foo": "bar"},
+			},
+		})
+
+		// select for namespaces with the label foo=bar
+		provisioner.Spec.NamespaceSelector = &metav1.LabelSelector{
+			MatchLabels: map[string]string{"foo": "bar"},
+		}
+		pod := ExpectProvisioned(ctx, env.Client, selectionController, provisioners, provisioner,
+			test.UnschedulablePod(test.PodOptions{ObjectMeta: metav1.ObjectMeta{Namespace: ns}}),
+		)[0]
+		ExpectScheduled(ctx, env.Client, pod)
+	})
+	It("should schedule if the pod is in a matching namespace, MatchExpressions", func() {
```

Comment:
Can we add a test to pick namespaces by name?

Reply:
Done